### PR TITLE
CORE-18000 - Capture the exception thrown by the state manager and abort the current batch of requests

### PR DESCRIPTION
**Pull request by filipesoliveira**

### CODE REVIEW

File: components/ledger/ledger-utxo-token-cache/src/main/kotlin/net/corda/ledger/utxo/token/cache/services/PerformanceClaimStateStoreImpl.kt [hunk 1]
```diff
@@ -69,9 +69,20 @@ class PerformanceClaimStateStoreImpl(
...
             )
 
-            val mismatchedState = stateManager.update(listOf(stateManagerState))
-                .map { it.value }
-                .firstOrNull()
+            val mismatchedState = try {
+               stateManager.update(listOf(stateManagerState))
+                    .map { it.value }
+                    .firstOrNull()
+            } finally {
```
Comment (vkolomeyko) [inline components/ledger/ledger-utxo-token-cache/src/main/kotlin/net/corda/ledger/utxo/token/cache/services/PerformanceClaimStateStoreImpl.kt]: I think the idea was to "capture the exception"... whereas `finally` block will be called always.
```suggestion
            } catch(ex: Exception) {
                logger.warn("Exception during execution of an update", ex)
```

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by owenstanford — APPROVED** [review]
LGTM

**Comment by filipesoliveira** [comment]
DON'T MERGE THIS PR FOR NOW.

**Pull request merged**